packages/gatsby/src/utils/worker/child/queries.ts

```diff
@@ -37,6 +38,10 @@ export async function saveQueriesDependencies(): Promise<void> {
     pickNecessaryQueryState
   )
 
+  // at this point, we're done grabbing page dependencies, so we need to
+  // flush out the batcher in case there are any left
+  createPageDependencyBatcher.flush()
+
```

Comment:
We need this to happen before `savePartialStateToDisk()` call otherwise we will persist before handling last batch of dependencies

Reply:
Per offline chat - let's move `.flush()` to `run*Queries` as that's more resilient